libs/util/httpRequest.js

```diff
@@ -9,154 +9,161 @@
 
 var FetchrError = require('./FetchrError');
 
-function shouldRetry(err, options, attempt) {
+function _shouldRetry(err) {
     if (err.reason === FetchrError.ABORT) {
         return false;
     }
 
-    if (attempt >= options.retry.maxRetries) {
+    if (this._currentAttempt >= this._options.retry.maxRetries) {
         return false;
     }
 
-    if (options.method === 'POST' && !options.retry.retryOnPost) {
+    if (this._options.method === 'POST' && !this._options.retry.retryOnPost) {
         return false;
     }
 
-    return options.retry.statusCodes.indexOf(err.statusCode) !== -1;
+    return this._options.retry.statusCodes.indexOf(err.statusCode) !== -1;
 }
 
-function delayPromise(fn, delay) {
+// _retry is the onReject promise callback that we attach to the
```

Comment:
Any thing to review in these changes or is it just reorganizing the code?

Reply:
Not really, just reorganizing.